### PR TITLE
[BACKLOG-30660] [BACKLOG-30582] [BACKLOG-30581] : CSRF - /api/system/refresh : /api/theme : /api/licenseManager

### PR DESCRIPTION
**Pull request by sVarma1992**
CSRF changes related to : backlog-30582 and backlog-30660

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 20m 16s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
assemblies/pentaho-solutions,user-console
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — CHANGES_REQUESTED** [review]
Please provide one PR per story, or, at a minimum, one commit per story.

Also, this PR is also including changes from BACKLOG-30581.

Additionally, please, like is the case of other request matchers in this file, leave an empty line between request matcher elements (between the request matcher element and the comment of the following one).

Thanks!

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 20m 20s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
assemblies/pentaho-solutions,user-console
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

File: assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/applicationContext-spring-security-csrf.xml [hunk 1]
```diff
@@ -40,14 +40,28 @@
...
+          POST api/theme/set
+         -->
+        <swm:regex-request-matcher pattern="^/api/theme/set\b.*" methods="POST" />
+
+        <!--
+          GET api/system/refresh/globalActions
+          GET api/system/refresh/metadata
+          GET api/system/refresh/systemSettings
+          GET api/system/refresh/reportingDataCache
+          GET api/system/refresh/mondrianSingleSchemaCache
```
Comment (dcleao) [inline assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/applicationContext-spring-security-csrf.xml:57]: Please add `GET /api/system/refresh/mondrianSchemaCache`

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by sVarma1992** [comment]
Will raise a new PR with the guidelines mentioned in the review

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 18m 53s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
assemblies/pentaho-solutions,user-console
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by sVarma1992** [comment]
Created a new PR with addressing the review comments

**Pull request closed without merging**